src/npc.cpp

```diff
@@ -285,18 +285,19 @@ void Npc::onRemoveCreature(Creature* creature, bool isLogout)
 	}
 }
 
-void Npc::onCreatureMove(Creature* creature, const Tile* newTile, const Position& newPos, const Tile* oldTile,
-                         const Position& oldPos, bool teleport)
+void Npc::onCreatureMove(std::shared_ptr<Creature> creature, std::shared_ptr<const Tile> newTile,
+                         const Position& newPos, std::shared_ptr<const Tile> oldTile, const Position& oldPos,
+                         bool teleport)
 {
 	Creature::onCreatureMove(creature, newTile, newPos, oldTile, oldPos, teleport);
 
-	if (creature == this || creature->getPlayer()) {
+	if (creature.get() == this || creature->getPlayer()) {
 		if (npcEventHandler) {
 			npcEventHandler->onCreatureMove(creature, oldPos, newPos);
 		}
 
-		if (creature != this) {
-			Player* player = creature->getPlayer();
+		if (creature.get() != this) {
+			auto player = creature->getPlayer();
```

Comment:
Could this be `const auto`
```suggestion
			const auto player = creature->getPlayer();
```